File: app/src/main/java/com/kobrakid/retroachievements/database/ConsoleDao.java
package com.kobrakid.retroachievements.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ConsoleDao {

    @Query("SELECT * FROM console")
    List<Console> getConsoleList();

    @Query("SELECT * FROM console WHERE id == :consoleID")
    List<Console> getConsoleWithID(int consoleID);

    @Query("DELETE FROM console")
    void clearTable();

    @Insert
    void insertConsole(Console console);

    @Update
    void updateConsole(Console console);

    @Delete
    void deleteConsole(Console console);

}

File: app/src/main/java/com/kobrakid/retroachievements/database/ConsoleDao.kt
package com.kobrakid.retroachievements.database

import androidx.room.*
import com.kobrakid.retroachievements.model.Console

@Dao
interface ConsoleDao {
    @get:Query("SELECT * FROM console")
    val consoleList: List<Console>

    @get:Query("SELECT * FROM console WHERE NumGames > 0")
    val nonEmptyConsoles: List<Console>

    @Query("SELECT * FROM console WHERE id == :consoleID")
    fun getConsoleWithID(consoleID: String): List<Console>

    @Query("DELETE FROM console")
    fun clearTable()

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertConsole(console: Console)

    @Update
    fun updateConsole(console: Console)

    @Delete
    fun deleteConsole(console: Console)
}